app/services/stake_pools/calculate_average_stats.rb

```diff
@@ -33,6 +33,10 @@ def call
 
     attr_reader :pool, :validator, :validator_ids, :sp_attrs
 
+    def delinquent?(score)
```

Comment:
Do not use method name that is the same as attribute name, but does a different thing. It's a bad practice. This method name should match its behaviour.